### PR TITLE
feat(#major); vesta-finance; Add Vesta Finance Subgraph

### PR DESCRIPTION
**Pull request by corerouter**
A test subgraph was deployed at https://thegraph.com/hosted-service/subgraph/corerouter/vesta-finance

### CODE REVIEW

File: subgraphs/vesta-finance/src/entities/market.ts [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+  market.inputTokenBalance = balanceAsset;
+  market.inputTokenPriceUSD = getCurrentAssetPrice(asset);
+  market.save();
+
+  getOrCreateMarketSnapshot(event, market);
+  getOrCreateMarketHourlySnapshot(event, market);
+  updateProtocolUSDLocked(event, netChangeUSD);
+}
+
+export function addMarketDepositVolume(
```
Comment (dmelotik) [inline subgraphs/vesta-finance/src/entities/market.ts:257]: is there another reason to have these explicit functions, aside from being able to call them from the event handlers? To me they seem unnecessary, and you could just call `addMarketVolume()` from event.ts

File: subgraphs/vesta-finance/src/entities/protocol.ts [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+
+  const marketHourlySnapshot = getOrCreateMarketHourlySnapshot(event, market);
+  marketHourlySnapshot.hourlySupplySideRevenueUSD =
+    marketHourlySnapshot.hourlySupplySideRevenueUSD.plus(revenueAmountUSD);
+  marketHourlySnapshot.hourlyTotalRevenueUSD =
+    marketHourlySnapshot.hourlyTotalRevenueUSD.plus(revenueAmountUSD);
+  marketHourlySnapshot.save();
+}
+
+export function addProtocolBorrowVolume(
```
Comment (dmelotik) [inline subgraphs/vesta-finance/src/entities/protocol.ts:194]: same thing here, I am curious what the design choice was. I may be missing something

File: subgraphs/vesta-finance/src/entities/market.ts [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+  if (!market) {
+    const id = asset.toHexString();
+    const inputToken = getOrCreateAssetToken(asset);
+    market = new Market(id);
+    market.protocol = getOrCreateLendingProtocol().id;
+    market.name = inputToken.name;
+    market.isActive = true;
+    market.canUseAsCollateral = true;
+    market.canBorrowFrom = true;
+    market.maximumLTV = MAXIMUM_LTV;
```
Comment (dmelotik) [inline subgraphs/vesta-finance/src/entities/market.ts:56]: is there a way to dynamically get this in vesta? seems to be different by different vaults on their frontend

File: subgraphs/vesta-finance/src/entities/market.ts (hunk 1 above)
Comment (corerouter) [reply subgraphs/vesta-finance/src/entities/market.ts]: The original implementation have different details functions for borrow.repay and so on. After TrueFi subgraph review, I merged their details into one concrete function. Just remove these middle layer function in latest commit at 6fd7ecf7a0f6bb9ed6d41b5e08b7f9206f0fbb19

File: subgraphs/vesta-finance/src/entities/protocol.ts (hunk 2 above)
Comment (corerouter) [reply subgraphs/vesta-finance/src/entities/protocol.ts]: Just removed these middle layer functions in commit at 4051ac8514ff8e59bd34e249b7a3342d8824c11b

File: subgraphs/vesta-finance/src/entities/market.ts (hunk 3 above)
Comment (corerouter) [reply subgraphs/vesta-finance/src/entities/market.ts]: Fixed at d85b20c7bda0a79872d3a08825b6b0e7fb7e3ba3 Thanks!

File: subgraphs/vesta-finance/src/entities/market.ts [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+    const maxLTV = setMaxLTV(id);
+    market = new Market(id);
+    market.protocol = getOrCreateLendingProtocol().id;
+    market.name = inputToken.name;
+    market.isActive = true;
+    market.canUseAsCollateral = true;
+    market.canBorrowFrom = true;
+    market.maximumLTV = maxLTV;
+    market.liquidationThreshold = maxLTV;
+    market.liquidationPenalty = LIQUIDATION_FEE_PERCENT;
```
[the remark this answers is not in the document]
Comment (corerouter) [reply subgraphs/vesta-finance/src/entities/market.ts]: Fixed at 8febc1e38f7f44ca5d0ddb276fc82673be68f244 Thanks!

[the remark this answers is not in the document]
Comment (dmelotik) [reply subgraphs/vesta-finance/src/entities/market.ts]: nice! one small thing, are they supposed to be .1%, if it is 10% it should return 10 for example not .1

[the remark this answers is not in the document]
Comment (corerouter) [reply subgraphs/vesta-finance/src/entities/market.ts]: Fixed at 41eb75cff4a225614024cb60c1e971a1c15584c7 Thanks a lot!

File: subgraphs/vesta-finance/src/entities/market.ts [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+} from "./protocol";
+import { getOrCreateAssetToken, getCurrentAssetPrice } from "./token";
+import { getOrCreateStableBorrowerInterestRate } from "./rate";
+import { EventType } from "./event";
+import { bigIntToBigDecimal, exponentToBigDecimal } from "../utils/numbers";
+import {
+  ACTIVE_POOL_CREATED_BLOCK,
+  ACTIVE_POOL_CREATED_TIMESTAMP,
+  BIGDECIMAL_HUNDRED,
+  BIGDECIMAL_ONE,
```
[the remark this answers is not in the document]
Comment (corerouter) [reply subgraphs/vesta-finance/src/entities/market.ts]: Fix at 191611374096e4dfc1f036b6620d9263ee841e67 Thanks!

File: deployment/deployment.json [hunk 6]
```diff
@@ -2001,6 +2001,35 @@
...
     }
   },
+  "vesta-finance": {
+    "schema": "lending",
+    "base": "vesta-finance",
+    "protocol": "vesta-finance",
+    "deployments": {
+      "vesta-finance-arbitrum": {
+        "network": "arbitrum",
+        "status": "dev",
```
Comment (dmelotik) [inline deployment/deployment.json:2011]: you can change this to prod now since it passed QA

**Review by dmelotik — APPROVED** [review]
Nice work @corerouter !!

File: deployment/deployment.json (hunk 6 above)
Comment (corerouter) [reply deployment/deployment.json:2011]: Sorry, overlook it yesterday.

**Comment by corerouter** [comment]
> Nice work @corerouter !!

Thank you very much for your help. @dmelotik

File: deployment/deployment.json (hunk 6 above)
Comment (dmelotik) [reply deployment/deployment.json:2011]: np

**Pull request merged**